packages/lesswrong/lib/sql/CreateIndexQuery.ts

```diff
@@ -57,16 +69,30 @@ class CreateIndexQuery<T extends DbObject> extends Query<T> {
    * beta (it would also be an awkward special case for us to use a constraint for this
    * instead of an index). A simple workaround is to simply coalesce where needed.
    */
-  private fieldNameToIndexField(fieldName: string) {
+  private fieldNameToIndexField(fieldName: string, isCaseInsensitive: boolean) {
     const type = this.table.getField(fieldName);
     const coalesceValue = type?.getIndexCoalesceValue();
     const tokens = fieldName.split(".");
-    const name = `"${tokens[0]}"`;
+    const name = this.getIndexFieldName(tokens);
+    const field = coalesceValue && this.isUnique
+      ? `COALESCE(${name}, ${coalesceValue})`
+      : name;
     return {
       useGin: tokens.length > 1 || this.table.getField(fieldName)?.isArray(),
-      field: coalesceValue && this.isUnique ? `COALESCE(${name}, ${coalesceValue})` : name,
+      field: isCaseInsensitive && type?.toConcrete() instanceof StringType
+        ? `LOWER(${field})`
+        : field,
     };
   }
+
+  private getIndexFieldName(tokens: string[]): string {
+    if (tokens.length < 1) {
+      throw new Error(`Invalid index field tokens: ${JSON.stringify(tokens)}`);
+    }
+    return tokens.length === 1 || this.getField(tokens[0])?.isArray()
+      ? `"${tokens[0]}"`
+      : `("${tokens[0]}"${tokens.slice(1).map((field) => `->'${field}'`).join("")})`;
```

Comment:
Could you add a comment here? I have no idea what this is trying to do.

Reply:
Looks like that map only works when tokens is of length 2?

Reply:
It works when length is _at least_ 2